pkg/tfcli/refresh.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tfcli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/crossplane-contrib/terrajet/pkg/tfcli/types"
+)
+
+// Refresh updates local state of the Cloud resource in a synchronous manner
+func (c *client) Refresh(ctx context.Context, id string) (RefreshResult, error) {
+	if id == "" && len(c.state.tfState) == 0 {
+		return RefreshResult{}, errors.New(errNoID)
+	}
+
+	if _, err := c.initConfiguration(types.OperationRefresh, false); err != nil {
+		return RefreshResult{}, err
+	}
+
+	var result RefreshResult
+	var err error
+	if len(c.state.tfState) == 0 {
+		result, err = c.importResource(ctx, id)
```

Comment:
I prefer to rely on [terraform import](https://www.terraform.io/docs/cli/import/index.html) in `tfcli` instead of crafting a tfState with an ID ourselves, as I'm inclined to keep things as close to "Terraform way" as possible for the sake of robustness. For importing new resources Terraform CLI's import command seems to be the documented and suggested way.